tree/ntuple/v7/doc/architecture.md

```diff
@@ -3,7 +3,439 @@ RNTuple Code Architecture
+  3) A locator format: how are byte ranges addressed (e.g., through an offset in a file or an object ID)
+
+That means that new backends are likely to have implications on the RNTuple format specification.
+
+The page sources and sinks are ROOT internal classes.
+They are not meant to be extended by users.
+
+Multi-Threading
+---------------
+
+The following options exist in RNTuple for multi-threaded data processing.
+
+### Implicit Multi-Threading
+When `ROOT::EnableImplicitMT()` is used, RNTuple uses ROOT's task arena to compress and decompress pages.
+That requires writes to be buffered and reads uses the cluster pool resp.
+The RNTuple data source for RDataFrame lets RDataFrame full control of the thread pool.
+That means that RDataFrame uses a separate data source for every thread, each of the data sources runs in serial mode.
+
+### Concurrent Readers
+Multiple readers and read the same RNTuple concurrently as long as access to every individual reader is sequential.
+
+### Parallel REntry Preparation
+Multiple `REntry` object can be concurrently prepared by multiple threads.
+I.e., construction and binding of the objects can happen in parallel.
+The actual reading and writing of entries (`RNTupleReader::LoadEntry()`, `RNTupleWriter::Fill()`) needs to be protected by a mutex.
+This is considered "mild scalability parallelization" in RNTuple.
+
+### RNTupleParallelWriter
+The parallel writer offers the most scalable writing interface.
+Multiple _fill contexts_ can concurrently serialize and compress data.
+Every fill context prepares a set of entire clusters in the final on-disk layout.
+When a fill context flushes data,
+a brief serialization point handles the RNTuple meta-data updates and the reservation of disk space to write into.
+
+Relationship to other ROOT components
+-------------------------------------
+
+The RNTuple classes have the following relationship to other parts of ROOT.
+
+The RNTuple classes use core ROOT infrastructure classes, such as error handling and logging.
+When necessary, RNTuple uses a `TFile` for reading and writing.
+The cases of writing to a local file and reading from a local file, a file from XRootD or from HTTP, do _not_ require `TFile`.
+For these cases, RNTuple depends on the `RRawFile` class and its XRootD and Davix plugins.
+
+For user-defined classes as well as sets and maps, RNTuple uses `TClass`.
+Simple types and other stdlib classes are natively supported and do not require dictionaries.
+
+Integration to RDataFrame is provided through an RNTuple data source.
+A universal RDataFrame constructor can create a data frame from either a TTree or an RNTuple with the same syntax.
+
+The RBrowser uses RNTuple classes to display RNTuple dataset information.
+
+Future Features
+---------------
+
+The following feature a planned for after the first RNTuple production version:
+  - RNTupleMT: advanced RNTupleReader that allows for free combination of chains and (indexed) friends
```

Comment:
Is this the new "processor" that we talked about?

Reply:
Yep, should be called RNTupleProcessor